### PR TITLE
refactor(stream,agg): pass all columns from stream chunk to managed state

### LINKED ISSUE #4185: Pass all columns from upstream chunk to agg states' `apply_batch`
Since we have `args` and `order_pairs` which contains the upstream column indices in `risingwave_stream::executor::aggregation::agg_call::AggCall`, and `AggCall`s will be passed into `new` when creating `ManagedTableState` implementations, these agg states can store the upstream column indices in their struct. However, currently in `GlobalSimpleAggExecutor` and `HashAggExecutor`, we select several columns from upstream chunk then pass them to `agg_state.apply_batch`, this makes agg states hard to find interested columns according to saved indices. Agg states like `GenericExtremeState` now determine the input columns by hardcoded inference logic, when bugs appear, it's very hard to debug.

I propose that we just pass all the columns to `agg_state.apply_batch`, this will make agg states easy to write and debug. I've done this for newly added `ManagedStringAggState`, the code is much simpler.

### PR DESCRIPTION
**Pull request by richardchien**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Fixes #4185.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

#4185

### CODE REVIEW

File: src/stream/src/executor/managed_state/aggregation/mod.rs [hunk 1]
```diff
@@ -143,19 +143,19 @@ pub enum ManagedStateImpl<S: StateStore> {
...
 
 impl<S: StateStore> ManagedStateImpl<S> {
-    pub async fn apply_batch(
+    pub async fn apply_chunk(
         &mut self,
         ops: Ops<'_>,
         visibility: Option<&Bitmap>,
-        data: &[&ArrayImpl],
+        columns: &[&ArrayImpl],
         epoch: u64,
```
Comment (BugenZhao) [inline src/stream/src/executor/managed_state/aggregation/mod.rs:151]: Looks like the `epoch` is never used now? 🤔

File: src/stream/src/executor/global_simple_agg.rs [hunk 2]
```diff
@@ -138,19 +137,7 @@ impl<S: StateStore> GlobalSimpleAggExecutor<S> {
     ) -> StreamExecutorResult<()> {
         let capacity = chunk.capacity();
         let (ops, columns, visibility) = chunk.into_inner();
-
-        // --- Retrieve all aggregation inputs in advance ---
-        let all_agg_input_arrays = agg_input_array_refs(agg_calls, &columns);
-        let pk_input_arrays = pk_input_array_refs(input_pk_indices, &columns);
```
Comment (BugenZhao) [inline src/stream/src/executor/global_simple_agg.rs:144]: Can we also remove these two utility functions?

**Review by BugenZhao — APPROVED** [review]
LGTM!

File: src/stream/src/executor/managed_state/aggregation/mod.rs (hunk 1 above)
Comment (richardchien) [reply src/stream/src/executor/managed_state/aggregation/mod.rs:151]: yep, it never reads from state table in `apply_chunk`

File: src/stream/src/executor/global_simple_agg.rs (hunk 2 above)
Comment (richardchien) [reply src/stream/src/executor/global_simple_agg.rs:144]: yes we can, removed in the latest commit

**Pull request merged**